### PR TITLE
Apply PHP 8.0 Syntax and constructor promotion

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 8.0, php 8.0 syntax can constuctor promotion can be applied.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

File: src/Service/ServiceManagerConfig.php [hunk 1]
```diff
@@ -64,37 +64,29 @@ class ServiceManagerConfig extends Config
...
-            'EventManagerAwareInitializer' => function ($first, $second) {
+            'EventManagerAwareInitializer' => static function ($first, $second) : void {
                 if ($first instanceof ContainerInterface) {
                     $container = $first;
                     $instance = $second;
                 } else {
                     $container = $second;
                     $instance = $first;
                 }
-
```
Comment (Ocramius) [inline src/Service/ServiceManagerConfig.php:84]: I really don't like that rector gets rid of random whitespace, but it's not critical :)

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

**Comment by Ocramius** [comment]
Thanks @samsonasik!

File: src/Service/ServiceManagerConfig.php (hunk 1 above)
Comment (samsonasik) [reply src/Service/ServiceManagerConfig.php:84]: that's on PHP-Parser latest dev-master for keep the space, but probably only will land on PHP-Parser 5.0 https://github.com/nikic/PHP-Parser/pull/891

Comment (Ocramius) [reply src/Service/ServiceManagerConfig.php:84]: Cool!

**Comment by TomasVotruba** [comment]
Amazing work :partying_face:

**Comment by Ocramius** [comment]
@TomasVotruba we still need to automate @samsonasik away xD

**Comment by TomasVotruba** [comment]
@Ocramius We're working on that together... :wink:  You'll notice our Github accounts start to look like human speaking :laughing:

**Comment by Ocramius** [comment]
@TomasVotruba aware, just saying that we probably need to put some effort in making this some sort of CI / bot.

Absolutely hats off to @samsonasik's thankless and constant efforts, but I think that would be a big win for the ecosystem at large :+1:

**Comment by TomasVotruba** [comment]
@Ocramius We already use it in CI as automated contributor: https://github.com/rectorphp/rector-src/blob/main/.github/workflows/rector.yaml
Just change the `rector.php` setup and everything else happens by itself :)

**Comment by Ocramius** [comment]
Yeh, I discussed it with @samsonasik before: what we need is something that **suggests** changes (with github actions output format), rather than something that applies the changes :)

I haven't worked on it myself yet, but probably something for https://github.com/rectorphp/rector/tree/b9c04e2a17bf233037d76ea2b9b2d7b411b5ec41/src/Console/Formatter

Someday I'll try helping out there :D

**Pull request merged**